app/server/src/db/userStore.ts

```diff
@@ -21,9 +21,19 @@ export class UserStore {
     private _sampleId = (sampleHash: string, fileName: string) => `${sampleHash}:${fileName}`;
 
     private async _setProjectName (projectId: string, newProjectName: string) {
+        await this._validProject(projectId);
         await this._redis.hset(this._projectKey(projectId), "name", newProjectName);
     }
 
+    private async _validProject(projectId: string): Promise<Record<string, unknown>> {
+        const project = await this._redis.hgetall(this._projectKey(projectId));
+        if (project.deleted_at) {
+            throw new Error(`This project has been deleted.`);
```

Comment:
If you throw a `BeebopError` here instead, you can include a message and status code that will make its way back to the client in the response.